pandas/core/algorithms.py

```diff
@@ -948,10 +948,39 @@ def select_n_frame(frame, columns, n, method, keep):
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
-    ser = getattr(frame[columns[0]], method)(n, keep=keep)
-    if isinstance(ser, Series):
-        ser = ser.to_frame()
-    return ser.merge(frame, on=columns[0], left_index=True)[frame.columns]
+    ascending = method == 'nsmallest'
+    original_index = frame.index
+    frame = frame.reset_index(drop=True)
```

Comment:
Ah, sorry @jreback, I was under the impression that without the inplace, ``reset_index`` would return a copy of the dataframe.
But it actually just returns a reference to the same dataframe with the resetted index?